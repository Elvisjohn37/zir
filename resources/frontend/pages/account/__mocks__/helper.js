let mockedModule = jest.createMockFromModule('../helper.js');

mockedModule.openAccountWindow = jest.fn();

module.exports = mockedModule;
