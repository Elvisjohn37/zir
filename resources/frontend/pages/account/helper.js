export function openAccountWindow(url) {
	window.open(url, 'account_window', 'width = 670, height = 800');
}
