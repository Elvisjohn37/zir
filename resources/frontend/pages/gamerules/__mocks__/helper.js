let mockedModule = jest.createMockFromModule('../helper');

mockedModule.extractProductGameType = (gameRulesList) => {
	let extractedItems = [];
	let indexMap = {};
	let nextIndex = 0;

	gameRulesList.forEach((item) => {
		if (item.productName in indexMap) {
			if (!extractedItems[indexMap[item.productName]].gameTypes.includes(item.gameTypeName)) {
				extractedItems[indexMap[item.productName]].gameTypes.push(item.gameTypeName);
			}
		} else {
			indexMap[item.productName] = nextIndex;
			nextIndex++;

			extractedItems.push({
				productName: item.productName,
				gameTypes: [item.gameTypeName],
			});
		}
	});

	return extractedItems;
};

module.exports = mockedModule;
