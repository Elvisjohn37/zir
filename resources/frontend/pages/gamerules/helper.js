import lodash from 'lodash';

function extractProductGameType(gameRulesList) {
	let extractedItems = [];
	let indexMap = {};
	let nextIndex = 0;

	gameRulesList.forEach((item) => {
		if (item.productName in indexMap) {
			if (!lodash.includes(extractedItems[indexMap[item.productName]].gameTypes, item.gameTypeName)) {
				extractedItems[indexMap[item.productName]].gameTypes.push(item.gameTypeName);
			}
		} else {
			indexMap[item.productName] = nextIndex;
			nextIndex++;

			extractedItems.push({
				productName: item.productName,
				gameTypes: [item.gameTypeName],
			});
		}
	});

	return extractedItems;
}

export { extractProductGameType };
