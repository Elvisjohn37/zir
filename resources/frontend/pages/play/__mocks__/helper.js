let postMessageHandler = () => jest.fn();

let setPostMessageHandler = () => jest.fn();

let resetPostMessageHandler = () => jest.fn();

resetPostMessageHandler();

// _________________________________________

let windowResponseData,
	isMobile = false;

let windowResponse = () => windowResponseData;

let resetWindowResponse = () => {
	windowResponseData = {
		result: true,
		data: {
			url: 'https://eyecon.classicku.com/launch/lek?uid=ANqWb0TXAFr1%252BZT7aQmY%252F1M%253D&guid=124275588462ff2e5f3973e3.1098258&gameid=71656&brand=LEK&cur=ZWD&lang=en&homeURL=https%3A%2F%2Fgames.classicku.com%2Fclosegame%3FgameID%3DDVbr%26clientID%3DML641',
			title: 'Beat The Bobbies',
			maxPayout: '700000000.000000',
			isMobile: false,
			viewport: 'desktop',
			productID: 'zy',
			balanceInfo: { playableBalance: '1457.800000', currencyCode: 'ZWD' },
		},
	};
};

let setWindowResponse = (data) => {
	windowResponseData = {
		...windowResponseData,
		data: {
			...windowResponseData.data,
			...data,
		},
	};
};

let setErrorResponse = () => {
	windowResponseData = {
		...windowResponseData,
		result: false,
	};
};

resetWindowResponse();

// _________________________________________

module.exports = { postMessageHandler, windowResponse, setWindowResponse, resetWindowResponse, setErrorResponse };
