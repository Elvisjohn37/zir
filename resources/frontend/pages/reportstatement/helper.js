import { getStatementReport as getStatementReportAjax } from 'frontend/ajax/backend';

export function getStatementReport(
	isMounted,
	setDateRange,
	statementRange,
	setStatementData,
	setStatementDataLoading,
	setStatementDataError
) {
	setStatementDataLoading(true);

	getStatementReportAjax(
		statementRange,
		!isMounted,
		(response) => {
			setStatementDataLoading(false);
			setStatementData({
				content: response.data.data.content,
				footer: response.data.data.footer,
			});
			if (!isMounted) {
				setDateRange({
					isLoading: false,
					range: response.data.data.dateRange,
				});
			}
		},
		() => {
			setStatementDataError(true);
			setStatementDataLoading(false);
		}
	);
}
