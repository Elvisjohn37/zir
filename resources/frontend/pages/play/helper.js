import { windowResponse } from 'frontend/utils/helper';

let postMessageHandlerWrapper = () => {
	let allowedOrigin = windowResponse() ? windowResponse().data.url : '';

	return function ({ data, origin }) {
		let originUrl = new URL(origin);
		let gameUrl = new URL(allowedOrigin);

		if ((gameUrl.origin == originUrl.origin || window.location.host == originUrl.host) && data == 'closeGame') {
			window.location.href = '/';
		}
	};
};

let postMessageHandler = postMessageHandlerWrapper();

export { postMessageHandler, windowResponse };
