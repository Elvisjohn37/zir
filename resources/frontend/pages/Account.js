import React from 'react';
import styles from './Account.module';
import Button from '@mui/material/Button';
import { LangDataProvider } from 'frontend/components/Language';
import { FormattedMessage } from 'react-intl';
import { Link } from 'react-router-dom';
import Grid from '@mui/material/Grid';
import { openAccountWindow } from './account/helper';
import { UserConsumer } from 'frontend/contexts/User';

import { AccountBalanceWallet, SwapHoriz, PanTool, RemoveCircle, Archive } from '@mui/icons-material';

export default function Account() {
	return (
		<LangDataProvider category="submenu">
			<UserConsumer>
				{({ userState }) => {
					return (
						<div className={styles.account}>
							<div className={styles.menuContainer}>
								<Grid container spacing={parseInt(styles.account_menuSpacing)} alignItems="center">
									{userState.user.isWalkIn === 1 && (
										<Grid item md={6} xs={6} lg={4}>
											<Button
												className={styles.label}
												onClick={() => {
													openAccountWindow('/account/transferfunds');
												}}
												classes={{ root: styles.accMenuBtn }}
												variant="contained"
												disableElevation
												color="primary"
											>
												<SwapHoriz fontSize="large" />
												<FormattedMessage id="transferfunds" />
											</Button>
										</Grid>
									)}
									<Grid item md={6} xs={6} lg={4}>
										<Button
											className={styles.label}
											onClick={() => {
												openAccountWindow('/account/balance');
											}}
											classes={{ root: styles.accMenuBtn, label: styles.label }}
											variant="contained"
											disableElevation
											color="primary"
										>
											<AccountBalanceWallet fontSize="large" />
											<FormattedMessage id="balance" />
										</Button>
									</Grid>
									<Grid item md={6} xs={6} lg={4}>
										<Button
											className={styles.label}
											component={Link}
											to="/account/archived-report"
											classes={{ root: styles.accMenuBtn, label: styles.label }}
											variant="contained"
											disableElevation
											color="primary"
										>
											<Archive fontSize="large" />
											<FormattedMessage id="archivedReport" />
										</Button>
									</Grid>
									<Grid item md={6} xs={6} lg={4}>
										<Button
											className={styles.label}
											onClick={() => {
												openAccountWindow('/account/limitadjust');
											}}
											classes={{ root: styles.accMenuBtn, label: styles.label }}
											variant="contained"
											disableElevation
											color="primary"
										>
											<RemoveCircle fontSize="large" />
											<FormattedMessage id="limitadjust" />
										</Button>
									</Grid>
									<Grid item md={6} xs={6} lg={4}>
										<Button
											className={styles.label}
											onClick={() => {
												openAccountWindow('/account/selfexclusion');
											}}
											classes={{ root: styles.accMenuBtn, label: styles.label }}
											variant="contained"
											disableElevation
											color="primary"
										>
											<PanTool fontSize="large" />
											<FormattedMessage id="selfexclusion" />
										</Button>
									</Grid>
								</Grid>
							</div>
						</div>
					);
				}}
			</UserConsumer>
		</LangDataProvider>
	);
}
