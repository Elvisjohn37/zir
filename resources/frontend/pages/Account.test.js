import React from 'react';
import { render, screen, cleanup, within } from '@testing-library/react';
import Account from './Account';
import userEvent from '@testing-library/user-event';
import * as OpenWindow from './account/helper';
import { setUserState } from 'frontend/contexts/User';

jest.mock('@mui/material/Button');
jest.mock('@mui/material/Grid');
jest.mock('frontend/components/Language');
jest.mock('frontend/contexts/User');
jest.mock('react-intl');
jest.mock('react-router-dom');
jest.mock('./account/helper');

describe('<Account />', () => {
	afterEach(() => {
		cleanup();
		jest.clearAllMocks();
		jest.restoreAllMocks();
	});

	it('should match snapshot', () => {
		setUserState({ user: { parent: 'test', isWalkIn: 1 } });
		let { asFragment } = render(<Account />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('should open window once clicked the transfer funds menu', () => {
		jest.spyOn(OpenWindow, 'openAccountWindow');
		setUserState({ user: { parent: 'test', isWalkIn: 1 } });
		render(<Account />);

		let button = screen.getAllByTestId('Button')[0];
		userEvent.click(button);
		expect(OpenWindow.openAccountWindow).toHaveBeenCalledWith('/account/transferfunds');
	});

	it('should open window once clicked the balance menu', () => {
		jest.spyOn(OpenWindow, 'openAccountWindow');
		setUserState({ user: { parent: 'test', isWalkIn: 1 } });
		render(<Account />);

		let button = screen.getAllByTestId('Button')[1];
		userEvent.click(button);
		expect(OpenWindow.openAccountWindow).toHaveBeenCalledWith('/account/balance');
	});

	it('should open window once clicked the limit adjust menu', () => {
		jest.spyOn(OpenWindow, 'openAccountWindow');
		setUserState({ user: { parent: 'test', isWalkIn: 1 } });
		render(<Account />);

		let button = screen.getAllByTestId('Button')[2];
		userEvent.click(button);
		expect(OpenWindow.openAccountWindow).toHaveBeenCalledWith('/account/limitadjust');
	});

	it('should open window once clicked the self exclusion menu', () => {
		jest.spyOn(OpenWindow, 'openAccountWindow');
		setUserState({ user: { parent: 'test', isWalkIn: 1 } });
		render(<Account />);

		let button = screen.getAllByTestId('Button')[3];
		userEvent.click(button);
		expect(OpenWindow.openAccountWindow).toHaveBeenCalledWith('/account/selfexclusion');
	});

	it('should redirect page once clicked the archived report', () => {
		setUserState({ user: { parent: 'test', isWalkIn: 1 } });
		let { container } = render(<Account />);

		expect(container.querySelector('a[to="/account/archived-report"]')).toBeInTheDocument();
	});

	it('should display funds menu if the player is walkin', () => {
		setUserState({ user: { parent: 'test', isWalkIn: 1 } });
		let { rerender } = render(<Account />);

		expect(screen.getByText('transferfunds')).toBeInTheDocument();

		setUserState({ user: { parent: 'test', isWalkIn: 0 } });
		rerender(<Account />);
		expect(screen.queryByText('transferfunds')).toBeNull();
	});
});
