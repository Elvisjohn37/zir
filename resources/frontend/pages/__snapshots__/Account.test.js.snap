// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Account /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    category="submenu"
    data-testid="LangDataProvider"
  >
    <div>
      submenu
    </div>
    <div
      class="account"
    >
      <div
        class="menuContainer"
      >
        <div
          alignitems="center"
          container="true"
          data-testid="Grid"
          spacing="0"
        >
          <div
            data-testid="Grid"
            item="true"
            lg="4"
            md="6"
            xs="6"
          >
            <button
              class="label"
              classes="{\\"root\\":\\"accMenuBtn\\"}"
              color="primary"
              data-testid="Button"
              disableelevation="true"
              variant="contained"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                data-testid="SwapHorizIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6.99 11 3 15l3.99 4v-3H14v-2H6.99v-3zM21 9l-3.99-4v3H10v2h7.01v3L21 9z"
                />
              </svg>
              <div
                data-testid="FormattedMessage"
                id="transferfunds"
              >
                <div>
                  transferfunds
                </div>
              </div>
            </button>
          </div>
          <div
            data-testid="Grid"
            item="true"
            lg="4"
            md="6"
            xs="6"
          >
            <button
              class="label"
              classes="{\\"root\\":\\"accMenuBtn\\",\\"label\\":\\"label\\"}"
              color="primary"
              data-testid="Button"
              disableelevation="true"
              variant="contained"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                data-testid="AccountBalanceWalletIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21 18v1c0 1.1-.9 2-2 2H5c-1.11 0-2-.9-2-2V5c0-1.1.89-2 2-2h14c1.1 0 2 .9 2 2v1h-9c-1.11 0-2 .9-2 2v8c0 1.1.89 2 2 2h9zm-9-2h10V8H12v8zm4-2.5c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
                />
              </svg>
              <div
                data-testid="FormattedMessage"
                id="balance"
              >
                <div>
                  balance
                </div>
              </div>
            </button>
          </div>
          <div
            data-testid="Grid"
            item="true"
            lg="4"
            md="6"
            xs="6"
          >
            <a
              class="label"
              classes="{\\"root\\":\\"accMenuBtn\\",\\"label\\":\\"label\\"}"
              color="primary"
              data-testid="Link"
              disableelevation="true"
              to="/account/archived-report"
              variant="contained"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                data-testid="ArchiveIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m20.54 5.23-1.39-1.68C18.88 3.21 18.47 3 18 3H6c-.47 0-.88.21-1.16.55L3.46 5.23C3.17 5.57 3 6.02 3 6.5V19c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V6.5c0-.48-.17-.93-.46-1.27zM12 17.5 6.5 12H10v-2h4v2h3.5L12 17.5zM5.12 5l.81-1h12l.94 1H5.12z"
                />
              </svg>
              <div
                data-testid="FormattedMessage"
                id="archivedReport"
              >
                <div>
                  archivedReport
                </div>
              </div>
            </a>
          </div>
          <div
            data-testid="Grid"
            item="true"
            lg="4"
            md="6"
            xs="6"
          >
            <button
              class="label"
              classes="{\\"root\\":\\"accMenuBtn\\",\\"label\\":\\"label\\"}"
              color="primary"
              data-testid="Button"
              disableelevation="true"
              variant="contained"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                data-testid="RemoveCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11H7v-2h10v2z"
                />
              </svg>
              <div
                data-testid="FormattedMessage"
                id="limitadjust"
              >
                <div>
                  limitadjust
                </div>
              </div>
            </button>
          </div>
          <div
            data-testid="Grid"
            item="true"
            lg="4"
            md="6"
            xs="6"
          >
            <button
              class="label"
              classes="{\\"root\\":\\"accMenuBtn\\",\\"label\\":\\"label\\"}"
              color="primary"
              data-testid="Button"
              disableelevation="true"
              variant="contained"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                data-testid="PanToolIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M23 5.5V20c0 2.2-1.8 4-4 4h-7.3c-1.08 0-2.1-.43-2.85-1.19L1 14.83s1.26-1.23 1.3-1.25c.22-.19.49-.29.79-.29.22 0 .42.06.6.16.04.01 4.31 2.46 4.31 2.46V4c0-.83.67-1.5 1.5-1.5S11 3.17 11 4v7h1V1.5c0-.83.67-1.5 1.5-1.5S15 .67 15 1.5V11h1V2.5c0-.83.67-1.5 1.5-1.5s1.5.67 1.5 1.5V11h1V5.5c0-.83.67-1.5 1.5-1.5s1.5.67 1.5 1.5z"
                />
              </svg>
              <div
                data-testid="FormattedMessage"
                id="selfexclusion"
              >
                <div>
                  selfexclusion
                </div>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
